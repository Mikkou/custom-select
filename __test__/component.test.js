import { mount } from '@vue/test-utils'
import CustomSelect from '../components/lib/customSelect'

const items = [
  { text: 'test1', value: 'val1' },
  { text: 'test2', value: 'val2' },
  { text: 'test3', value: 'val3' },
  { text: 'test4', value: 'val4' },
  { text: 'test5', value: 'val5' },
  { text: 'test6', value: 'val6' },
  { text: 'test7', value: 'val7' },
  { text: 'test8', value: 'val8' },
  { text: 'test9', value: 'val9' }
]

const value = {}

describe('CustomSelect', () => {
  test('check html', () => {
    const wrapper = mount(CustomSelect, {
      propsData: { items, value }
    })
    wrapper.find('.m-select__title').trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })
})
