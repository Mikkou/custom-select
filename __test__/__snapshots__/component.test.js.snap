// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect check html 1`] = `
<div
  class="m-select"
>
  <div
    class="m-select__title py-2 px-2"
  >
    
  
  </div>
   
  <div
    class="m-select__list"
    name="fade"
  >
    <ul>
      <li
        class="m-select__item px-2 py-2"
      >
        test1
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test2
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test3
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test4
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test5
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test6
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test7
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test8
        
      </li>
      <li
        class="m-select__item px-2 py-2"
      >
        test9
        
      </li>
    </ul>
  </div>
</div>
`;
